MetaDataset.py

```diff
@@ -968,7 +1058,7 @@ def _expand_dataset_sec_idxs(self, num_values):
 
   def _load_seqs(self, start, end):
     # If the segment order is not yet known, fix the next few segments
-    if end > len(self.dataset_sorted_seq_idx_list):
+    if not self._can_use_num_seqs and end > len(self.dataset_sorted_seq_idx_list):
```

Comment:
Why do you need this check?

Reply:
Because calling `self._expand_dataset_sec_idxs` does not work in case of `_can_use_num_seqs`. `used_num_seqs_per_subset` is not created in that case and also not all sequence orderings are covered in the function.

Reply:
Why don't you just create `used_num_seqs_per_subset` in all cases? It's bad that there is some attrib which is only used in some cases but not in others.